src/main/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaData.java

```diff
@@ -640,8 +644,315 @@ private void addImportedKey(final List<List<Object>> metaData,
 
     @Override
     public ResultSet getTypeInfo() throws SQLException {
-        // TODO: Implement
-        throw new SQLFeatureNotSupportedException();
+        int UNUSED = -1;
+        int BASE_2 = 2;
+        int BASE_10 = 10;
+        return new DocumentDbListResultSet(
+                null,
+                buildTypeInfoColumnMetaData(),
+                new ArrayList<>(Arrays.asList(
+                        new ArrayList<>(Arrays.asList(
+                                "BOOLEAN", // TYPE_NAME
+                                Types.BOOLEAN, // DATA_TYPE
+                                TYPE_COLUMN_SIZE_MAP.get(Types.BOOLEAN), // PRECISION
+                                null, // LITERAL_PREFIX
+                                null, // LITERAL_SUFFIX
+                                null, // CREATE_PARAMS
+                                ResultSetMetaData.columnNullable, // NULLABLE
+                                true, // CASE_SENSITIVE
+                                DatabaseMetaData.typeSearchable, // SEARCHABLE
+                                true, // UNSIGNED_ATTRIBUTE
```

Comment:
I believe all these integers and real types are SIGNED - which means these flags should be `false`?

Reply:
Good catch

Reply:
This change is still required.